File: src/server/Models/Task.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using server.Interfaces;

public class Task : IUpdatableItem, ISoftDeletable, IUpdatableItemColor
{
    public Guid Id { get; set; }
    [Required(ErrorMessage = "Заголовок обязателен для заполнения.")]
    [StringLength(100, MinimumLength = 1, ErrorMessage = "Заголовок должен содержать от 1 до 100 символов.")]
    public required string Title { get; set; }
    [StringLength(5000, ErrorMessage = "Описание не может превышать 5000 символов.")]
    public string? Description { get; set; }
    public bool CheckBox {get; set; }
    public bool IsFavorite{ get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime LastModifiedAt { get; set; }
    public bool IsTrashed { get; set; }
    [Required(ErrorMessage = "Цвет фона обязателен для заполнения.")]
    public BackgroundColor BackgroundColor { get; set; }
    public bool IsDeleted { get; set; } 
    public DateTime? DeletedAt { get; set; }

    // Внешние ключи 
    [ForeignKey("User")]
    [Required(ErrorMessage = "Идентификатор создателя обязателен для заполнения.")]
    public Guid CreatedBy { get; set; } 
    public Guid? TaskGroupId  {get; set; }

    // Навигационные свойства
    public User User { get; set; }
    public TaskGroup? TaskGroup { get; set; }
}

File: src/client/src/features/tasks/task.ts
import { BackgroundColor } from "@/shared/enums/backgroundColor";
import { User } from "@/shared/types/user";
import { TaskGroup } from "./taskGroup";

export interface GetTaskResponse {
  id: string;
  title: string;
  description: string | null;
  checkBox: boolean;
  isFavorite: boolean;
  createdAt: string;
  lastModifiedAt: string;
  isTrashed: boolean;
  backgroundColor: BackgroundColor;
  isDeleted: boolean;
  deletedAt: string | null;
  taskGroup: TaskGroup | null;
  createdBy: User;
}

export interface CreateTaskRequest {
  title: string;
  description?: string | null;
  checkBox: boolean;
  isFavorite: boolean;
  backgroundColor: number;
  taskGroupId?: string | null;
}

export interface CreateTaskResponse {
  id: string;
  title: string;
  description: string | null;
  checkBox: boolean;
  isFavorite: boolean;
  createdAt: string;
  lastModifiedAt: string;
  isTrashed: boolean;
  backgroundColor: BackgroundColor;
  isDeleted: boolean;
  deletedAt: string | null;
  taskGroupId: string | null;
  createdBy: string;
}

export interface ToggleFavoriteResponse {
  id: string;
  isFavorite: boolean;
  lastModifiedAt: string;
}

export interface ToggleCheckboxResponse {
  id: string;
  checkBox: boolean;
  lastModifiedAt: string;
}

export interface UpdateTaskTitleRequest {
  id: string;
  title: string;
}

export interface UpdateTaskDescriptionRequest {
  id: string;
  description: string;
}

export interface UpdateTaskResponse {
  id: string;
  title: string | null;
  description: string | null;
  lastModifiedAt: string;
  wasUpdated: boolean;
}
